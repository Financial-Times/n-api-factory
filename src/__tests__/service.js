import fetch from 'node-fetch';

import setupService from '../service';

const config = {
	API_HOST: 'http://mock.com',
	API_KEY: 'mock.key',
};

process.env.SYSTEM_CODE = 'newspaper-mma';

jest.mock('node-fetch', () =>
	jest.fn(() => ({
		ok: true,
		headers: {
			get: () => 5,
		},
		json: () => ({
			data: 'test',
		}),
	})),
);

describe('setupService', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('set up get method correctly', async () => {
		const service = setupService(config);
		const transactionId = 'xxxx';
		const res = await service.get({
			endpoint: '/test',
			meta: { transactionId },
			query: { foo: 'bar' },
		});
		expect(res).toEqual({ data: 'test' });
		expect(fetch.mock.calls).toMatchSnapshot();
	});

	it('set up post methods correctly', async () => {
		const service = setupService(config);
		const transactionId = 'xxxx';
		await service.post({
			endpoint: '/test',
			body: { test: 'test' },
			meta: { transactionId },
		});
		expect(fetch.mock.calls).toMatchSnapshot();
	});

	it('set up delete methods correctly', async () => {
		const service = setupService(config);
		const transactionId = 'xxxx';
		await service.delete({ endpoint: '/test/xxxx', meta: { transactionId } });
		expect(fetch.mock.calls).toMatchSnapshot();
	});
});
