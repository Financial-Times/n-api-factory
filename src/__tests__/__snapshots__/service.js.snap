// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupService set up delete methods correctly 1`] = `
Array [
  Array [
    "http://mock.com/test/xxxx",
    Object {
      "headers": Object {
        "Content-Type": "application/json",
        "FT-Transaction-Id": "xxxx",
        "X-Origin-System-Id": "https://cmdb.ft.com/systems/newspaper-mma",
        "X-Origin-User": "customer",
        "x-api-key": "mock.key",
      },
      "method": "DELETE",
    },
  ],
]
`;

exports[`setupService set up get method correctly 1`] = `
Array [
  Array [
    "http://mock.com/test?foo=bar",
    Object {
      "headers": Object {
        "Content-Type": "application/json",
        "FT-Transaction-Id": "xxxx",
        "X-Origin-System-Id": "https://cmdb.ft.com/systems/newspaper-mma",
        "X-Origin-User": "customer",
        "x-api-key": "mock.key",
      },
      "method": "GET",
    },
  ],
]
`;

exports[`setupService set up post methods correctly 1`] = `
Array [
  Array [
    "http://mock.com/test",
    Object {
      "body": "{\\"test\\":\\"test\\"}",
      "headers": Object {
        "Content-Type": "application/json",
        "FT-Transaction-Id": "xxxx",
        "X-Origin-System-Id": "https://cmdb.ft.com/systems/newspaper-mma",
        "X-Origin-User": "customer",
        "x-api-key": "mock.key",
      },
      "method": "POST",
    },
  ],
]
`;
